azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/collection/ResourcesInGroup.java

```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.resources.fluentcore.collection;
+
+public interface ResourcesInGroup<T, DefinitionT> extends
+    SupportsListing<T>,
+    SupportsCreating<DefinitionT>,
+    SupportsDeleting {}
```

Comment:
Generic way to get a any resource using id is:

resourceId =  ` "/subscriptions/<subid>/resourceGroups/<rgname>/providers/<resourceProviderNamespace>/<resourceType>/<resource-name>"`

```java
resources().list('resourceType eq <resourceProviderNamespace>/<resourceType> and name eq <resource-name>', null, null).get(0);
```

it returns an object of type `GenericResource` with properties as a plain Java object.